client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/configure-email-notification/email-address-editor.tsx

```diff
@@ -61,13 +68,43 @@ export default function EmailAddressEditor( {
 
 	const requestVerificationCode = useRequestVerificationCode();
 	const verifyEmail = useValidateVerificationCode();
+	const resendCode = useResendVerificationCode();
 
 	// Function to handle resending verification code
 	const handleResendCode = useCallback( () => {
-		setValidationError( undefined );
+		if ( emailItem.email ) {
+			setValidationError( undefined );
+			resendCode.mutate( { type: 'email', value: emailItem.email } );
+		}
+		// Disabled because we don't want to re-run this effect when resendCode changes
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ emailItem.email ] );
+
+	// Function to handle resend code button click
+	const handleResendCodeClick = useCallback( () => {
```

Comment:
Since we are using resend-verification logic when the user clicks on `Verify` or `Pending` after they have decided to verify "Later", we don't want to show the message in the UI when the resend button is clicked. Plus, I would like to record the event only when the `resend` button is clicked. Ths is why I separated the logic here.

Reply:
Thanks for clarifying this. make sense to me. 👍